segmentation/utils/data_utils_test.py

```diff
@@ -19,11 +20,36 @@ def _create_img_dir(temp_dir, fovs, imgs, img_sub_folder="TIFs"):
 
     for fov in fovs:
         fov_path = os.path.join(temp_dir, fov, img_sub_folder)
-        os.makedirs(fov_path)
+        if not os.path.exists(fov_path):
+            os.makedirs(fov_path)
         for img in imgs:
             io.imsave(os.path.join(fov_path, img), tif)
 
 
+def test_validate_paths():
+    valid_path = '.'
+
+    valid_parts = [p for p in pathlib.Path(valid_path).resolve().parts]
+    valid_parts[1] = 'not_a_real_directory'
+
+    starts_out_of_scope = os.path.join(*valid_parts)
+
+    entirely_out_of_scope = '/not_a_real_directory/somewhere_else_not_here'
+
+    nonexistant = './not_a_real_directory'
+
+    data_utils.validate_paths(valid_path)
+
+    with pytest.raises(ValueError, match=r".*starts*"):
```

Comment:
Woah, I didn't know this was possible. Super cool!

Reply:
oh yeah! pytest can catch/match exceptions with standard regex.  I was about to wrap an assert statement in a try/except structure, but then I found this